builtins/amp-ad.js

```diff
@@ -199,17 +200,18 @@ export function installAd(win) {
     isPositionFixed() {
       let el = this.element;
       const body = el.ownerDocument.body;
+      let hasFixedAncestor = false;
       do {
         if (POSITION_FIXED_TAG_WHITELIST[el.tagName]) {
           return false;
         }
         if (this.getWin()/*because only called from onLayoutMeasure */
             ./*OK*/getComputedStyle(el).position == 'fixed') {
-          return true;
+          hasFixedAncestor =  true;
```

Comment:
If it returns or breaks here, it won't detect that if this is inside a flying carpet. The `<div class="-amp-flying-carpet-container">` is the `position: fixed` element (and what'll return `true` here), not the `<amp-flying-carpet>` element.

So I have to continue going up the tree to see if it's a descendent of the flying carpet. Notice that this will not affect the "normal" case where an ad has no fixed ancestor (they always go up the tree to the body). This will make the "bad" case slower, since before we'd immediately return whereas now we'll continue up.